Library/Homebrew/cask/artifact/artifact.rb

```diff
@@ -19,17 +19,18 @@ def self.english_name
         "Generic Artifact"
       end
 
-      sig { params(cask: Cask, args: T.untyped).returns(T.attached_class) }
-      def self.from_args(cask, *args)
-        source, options = args
-
+      sig {
+        params(cask: Cask, source: T.nilable(T.any(String, Pathname)), options: T.untyped)
+          .returns(T.attached_class)
+      }
+      def self.from_args(cask, source = nil, **options)
         raise CaskInvalidError.new(cask.token, "No source provided for #{english_name}.") if source.blank?
 
         unless options.try(:key?, :target)
           raise CaskInvalidError.new(cask.token, "#{english_name} '#{source}' requires a target.")
         end
 
-        new(cask, source, **options)
+        new(cask, source, **T.unsafe(options))
```

Comment:
Same here: explicit/typed args would be nicer here.

Reply:
Cask DSL and artifacts are full of this. I absolutely agree this should change but that will be a larger refactor that I can open an issue tracking now.

I've always had the mindset to touch cask as little as possible unless it is its own dedicated PR with cask reviewers. Past experience has told me the design decisions of cask can be quite different than the rest of `brew`.

Reply:
Works for me, thanks @Bo98 👍🏻

Reply:
Managed to get rid of the cask artifact changes.